barretenberg/cpp/src/barretenberg/smt_verification/circuit/standard_circuit.cpp

```diff
@@ -545,8 +604,35 @@ size_t StandardCircuit::handle_shr_constraint(size_t cursor)
         STerm left = this->symbolic_vars[left_idx];
         STerm out = this->symbolic_vars[out_idx];
 
-        STerm shled = left >> nr.second;
-        out == shled;
+        // Simulate the shr circuit using bitwise ops
+        uint32_t shift = nr.second;
+        if ((shift & 1) == 1) {
```

Comment:
Where are these formulas from? An explanation wouldn't hurt. It is very hard to understand what's happening here without context